src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleFreeze.kt

```diff
@@ -35,23 +35,49 @@ object ModuleFreeze : Module("Freeze", Category.MOVEMENT) {
 
     private val disableOnFlag by boolean("DisableOnFlag", true)
 
+    private var velocityX = 0.0
+    private var velocityY = 0.0
+    private var velocityZ = 0.0
+    private var x = 0.0
+    private var y = 0.0
+    private var z = 0.0
+
+    override fun enable() {
+        velocityX = player.velocity.x
+        velocityY = player.velocity.y
+        velocityZ = player.velocity.z
+        x = player.x
+        y = player.y
+        z = player.z
+    }
+
+    override fun disable() {
+        player.velocity.x = velocityX
+        player.velocity.y = velocityY
+        player.velocity.z = velocityZ
+    }
+
     val moveHandler = handler<PlayerMoveEvent> { event ->
-        // Set motion to zero
         event.movement.x = 0.0
         event.movement.y = 0.0
         event.movement.z = 0.0
+        player.pos.x = x
+        player.pos.y = y
+        player.pos.z = z
     }
 
     val packetHandler = handler<PacketEvent> { event ->
-        if (mc.world != null && event.origin == TransferOrigin.RECEIVE) {
-            if (event.packet is PlayerPositionLookS2CPacket && disableOnFlag) {
-                enabled = false
+        if (mc.world != null) {
```

Comment:
World will never be `null` here, see `Module#209`.

Reply:
izuna had it there on the original freeze module

Reply:
It's still useless.

Reply:
Yes it is useless.